packages/scenes/src/variables/sets/SceneVariableSet.test.tsx

```diff
@@ -483,6 +485,51 @@ describe('SceneVariableList', () => {
       expect(C.state.loading).toBe(false);
     });
   });
+
+  describe('isVariableLoadingOrWaitingToUpdate', () => {
+    it('Should return true when loading or waiting to update', async () => {
+      const A = new TestVariable({ name: 'A', query: 'A.*', value: '', text: '', options: [] });
+      const B = new TestVariable({ name: 'B', query: 'A.$A', value: '', text: '', options: [] });
+
+      const set = new SceneVariableSet({ variables: [A, B] });
+      set.activate();
+
+      // Should start variables with no dependencies
+      expect(A.state.loading).toBe(true);
+      expect(B.state.loading).toBe(undefined);
+
+      expect(set.isVariableLoadingOrWaitingToUpdate(A)).toBe(true);
+      expect(set.isVariableLoadingOrWaitingToUpdate(B)).toBe(true);
+
+      A.signalUpdateCompleted();
+      expect(set.isVariableLoadingOrWaitingToUpdate(A)).toBe(false);
```

Comment:
Just for clarity maybe:
```suggestion
      expect(set.isVariableLoadingOrWaitingToUpdate(A)).toBe(false);
      expect(set.isVariableLoadingOrWaitingToUpdate(B)).toBe(true);
```